library/blivet.py

```diff
@@ -209,6 +209,8 @@ def _resize(self):
                 raise BlivetAnsibleError("volume '%s' cannot be resized from %s to %s: %s" % (self._device.name,
                                                                                               self._device.size,
                                                                                               size, str(e)))
+        elif size and self._device.size != size and not self._device.resizable:
+            raise BlivetAnsibleError("volume '%s' cannot be resized from %s to %s" % (self._device.name, self._device.size, size))
```

Comment:
Is it intended that the role fails in this case already in the "get required packages" task, before the actual resizing attempt?